yaksh/static/yaksh/js/requesthandler.js

```diff
@@ -0,0 +1,139 @@
+request_status = "initial"
+count = 0;
+checker = null
+function submitRequest(){
+    document.forms["code"].submit();
+}
+
+function check_state(state, uid) {
+    if ((state == "running" || state == "not started") && count < 7) {
+        count++;
+        setTimeout(function() {get_result(uid);}, 2000);
+    } else if (state == "unknown") {
+        request_status = "initial";
+        count = 0;
+        notify("Request timeout. Try again later");
+        clearInterval(checker);
+        unlock_screen();
+    } else {
+        request_status = "initial";
+        count = 0;
+        notify("Please try after few minutes");
+        clearInterval(checker);
+        unlock_screen();
+    }
+}
+
+function notify(text) {
+    var $notice = document.getElementById("notification");
+    $notice.classList.add("alert");
+    $notice.classList.add("alert-success");
+    $notice.innerHTML = text;
+}
+
+function lock_screen() {
+    document.getElementById("ontop").style.display = "block";
+}
+
+function unlock_screen() {
+    document.getElementById("ontop").style.display = "none";
+}
+
+function check_lock_screen() {
+    var $ontop_div = document.getElementById("ontop");
+    if ($ontop_div.style.display == "block") {
+        $ontop_div.style.display = "none";
```

Comment:
This is not really doing anything different from unlock_screen.  Check should just return a true/false, the name is therefore inconsistent with what the function is doing.

Reply:
Yes the the name is inconsistent!
Removed this function as it was a callback function for setInterval.